src/js/playlist/playlist.js

```diff
@@ -43,6 +45,14 @@ define([
                 item.feedid = item.feedid || feedid;
             }
 
+            if (item.feedInstanceId || feedInstanceId) {
+                item.feedInstanceId = item.feedInstanceId || feedInstanceId;
+            }
+
+            if (item.pinSetId || pinSetId) {
+                item.pinSetId = item.pinSetId || pinSetId;
+            }
```

Comment:
get rid of this `pinSetId` copying - because not all items in the playlist are pinned. This should already be in the item.